docs/source/assets_writer.rst

```diff
@@ -17,98 +17,112 @@
 
 .. _assets_writer:
 
-Assets writer
-=============
+Exploiting the map generated by Cartographer ROS
+================================================
 
-The purpose of SLAM is to compute the trajectory of a single sensor through a metric space.
-On a higher level, the input of SLAM is sensor data, its output is the best estimate of the trajectory up to this point in time.
-To be real-time and efficient, Cartographer throws most of the sensor data away immediately.
+As sensors data come in, the state of a SLAM algorithm such as Cartographer evolves to stay *the current best estimate* of a robot's trajectory and suroundings.
+The most accurate localization and mapping Cartographer can offer is therefore the one obtained when the algorithm finishes.
+Cartographer can serialize its internal state in a ``.pbstream`` file format which is essentially a compressed protobuf file containing a snapshot of the data structures used by Cartographer internally.
 
-The trajectory alone is rarely of interest.
-But once the best trajectory is established, the full sensor data can be used to derive and visualize information about its surroundings.
+To run efficiently in real-time, Cartographer throws most of its sensor data away immediately and only works with a small subset of its input, the mapping used internally (and saved in ``.pbstream`` files) is then very rough.
+However, when the algorithm finishes and a best trajectory is established, it can be recombined *a posteriori* with the full sensors data to create a high resolution map.
 
-Cartographer provides the assets writer for this.
-Its inputs are
+Cartographer makes this kind of recombination possible using ``cartographer_assets_writer``.
+The assets writer takes as input
 
-1. the original sensor data fed to SLAM in a ROS bag file,
-2. the cartographer state, which is contained in the ``.pbstream`` file that SLAM creates,
-3. the sensor extrinsics (i.e. TF data from the bag or a URDF),
+1. original sensors data that have been used to perform SLAM (in a ROS ``.bag`` file),
+2. a cartographer state captured while performing SLAM on those sensors data (in a ``.pbstream`` file),
+3. the sensor extrinsics (i.e. TF data from the bag or a URDF description),
 4. and a pipeline configuration, which is defined in a ``.lua`` file.
 
-The assets writer runs through the sensor data in batches with a known trajectory.
-It can be used to color, filter and export SLAM point cloud data in a variety of formats.
-For more information on what the asset writer can be used for, refer to the examples below below and the header files in `cartographer/io`_.
+The assets writer runs through the ``.bag`` data in batches with the trajectory found in the ``.pbstream``.
+The pipeline can be can be used to color, filter and export SLAM point cloud data into a variety of formats.
```

Comment:
The link to the points processors implementation should be kept in some form. E.g.:

```
There are multiple of such points processing steps that can be interleaved in a pipeline - several ones are already available from `cartographer/io`_.
```